x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -69,7 +69,13 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
 
     this.pendingCount$.next(this.pendingCount$.getValue() + 1);
 
+    this.deps.session.trackSearch(request, options?.sessionId);
     return this.runSearch(request, combinedSignal, strategy).pipe(
+      tapOnce((response) => {
+        if (response.id) {
```

Comment:
We don't need to do this when restoring a search, right?

```suggestion
        if (!id && response.id) {
```

Reply:
I didn't handle the `restore` use case in this PR yet, to keep it simple.